perception/run_perception_system.py

```diff
@@ -0,0 +1,202 @@
+from pydrake.systems.framework import DiagramBuilder
+from pydrake.util.eigen_geometry import Isometry3
+import pydrake.perception as mut
+
+import meshcat.transformations as tf
+from optimization_based_point_cloud_registration import (
+    AlignSceneToModel, ThresholdArray)
+from point_cloud_to_pose_system import PointCloudToPoseSystem
+
+
+def SegmentFoamBrick(scene_points, scene_colors):
+    """Removes all points that aren't a part of the foam brick.
+
+    @param scene_points An Nx3 numpy array representing a scene.
+    @param scene_colors An Nx3 numpy array of rgb values corresponding to the
+        points in scene_points.
+
+    @return brick_points An Mx3 numpy array of points in the brick.
+    @return brick_colors An Mx3 numpy array of the colors of the brick points.
+    """
+
+    x_min = 0.2
+    x_max = 1.0
+
+    y_min = -0.4
+    y_max = 0.4
+
+    z_min = -0.1
+    z_max = 0.1
+
+    x_indices = ThresholdArray(scene_points[:, 0], x_min, x_max)
+    y_indices = ThresholdArray(scene_points[:, 1], y_min, y_max)
+    z_indices = ThresholdArray(scene_points[:, 2], z_min, z_max)
+
+    indices = reduce(np.intersect1d, (x_indices, y_indices, z_indices))
+
+    table_points = scene_points[indices, :]
+    table_colors = scene_colors[indices, :]
+
+    r_min = 0.2
+    r_max = 1
+
+    g_min = 0
+    g_max = 0.15
+
+    b_min = 0
+    b_max = 0.15
+
+    r_indices = ThresholdArray(table_colors[:, 0], r_min, r_max)
+    g_indices = ThresholdArray(table_colors[:, 1], g_min, g_max)
+    b_indices = ThresholdArray(table_colors[:, 2], b_min, b_max)
+
+    indices = reduce(np.intersect1d, (r_indices, g_indices, b_indices))
+
+    brick_points = table_points[indices, :]
+    brick_colors = table_colors[indices, :]
+
+    return brick_points, brick_colors
+
+def GetFoamBrickPose(brick_points, brick_colors):
+    """Finds a good 4x4 pose of the brick from the segmented points.
+
+    @param brick_points An Nx3 numpy array of brick points.
+    @param brick_colors An Nx3 numpy array of corresponding brick colors.
+
+    @return X_MS A 4x4 numpy array of the best-fit brick pose.
+    """
+    model_brick = np.load("models/foam_brick_model.npy")
+
+    num_sample_points = min(brick_points.shape[0], 250)
+    X_MS, error = AlignSceneToModel(
+        brick_points, model_brick, num_sample_points=num_sample_points)
+
+    # if the best fit matrix isn't exactly an Isometry3, fix it
+    try:
+        Isometry3(X_MS)
+    except:
+        # make valid Isometry3
+        sin_th = X_MS[1, 0]
+        cos_th = X_MS[0, 0]
+
+        if sin_th > 0:
+            theta = np.arccos(np.clip(cos_th, -1.0, 1.0))
+        else:
+            theta = -np.arccos(np.clip(cos_th, -1.0, 1.0))
+
+        X_MS[0, 0] = np.cos(theta)
+        X_MS[1, 1] = np.cos(theta)
+        X_MS[0, 1] = -np.sin(theta)
+        X_MS[1, 0] = np.sin(theta)
+
+    return X_MS
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
```

Comment:
It seems that this argument is not optional? You could add something like `required=True` as an argument to `parser.add_argument`.